library/src/main/java/com/bumptech/glide/load/data/StreamLocalUriFetcher.java

```diff
@@ -12,17 +19,78 @@
  * Fetches an {@link java.io.InputStream} for a local {@link android.net.Uri}.
  */
 public class StreamLocalUriFetcher extends LocalUriFetcher<InputStream> {
+    /** A lookup uri (e.g. content://com.android.contacts/contacts/lookup/3570i61d948d30808e537) */
+    private static final int ID_LOOKUP = 1;
+    /** A contact thumbnail uri (e.g. content://com.android.contacts/contacts/38/photo) */
+    private static final int ID_THUMBNAIL = 2;
+    /** A contact uri (e.g. content://com.android.contacts/contacts/38) */
+    private static final int ID_CONTACT = 3;
+    /**
+     * A contact display photo (high resolution) uri
+     * (e.g. content://com.android.contacts/5/display_photo)
+     */
+    private static final int ID_DISPLAY_PHOTO = 4;
+
+
+    private static final UriMatcher URI_MATCHER;
+
+    static {
+        URI_MATCHER = new UriMatcher(UriMatcher.NO_MATCH);
+        URI_MATCHER.addURI(ContactsContract.AUTHORITY, "contacts/lookup/*/#", ID_LOOKUP);
+        URI_MATCHER.addURI(ContactsContract.AUTHORITY, "contacts/lookup/*", ID_LOOKUP);
+        URI_MATCHER.addURI(ContactsContract.AUTHORITY, "contacts/#/photo", ID_THUMBNAIL);
+        URI_MATCHER.addURI(ContactsContract.AUTHORITY, "contacts/#", ID_CONTACT);
+        URI_MATCHER.addURI(ContactsContract.AUTHORITY, "contacts/#/display_photo", ID_DISPLAY_PHOTO);
+    }
+
     public StreamLocalUriFetcher(Context context, Uri uri) {
         super(context, uri);
     }
 
     @Override
     protected InputStream loadResource(Uri uri, ContentResolver contentResolver) throws FileNotFoundException {
+        if (ContactsContract.Contacts.CONTENT_URI.getHost().equals(uri.getHost())
+                && URI_MATCHER.match(uri) != UriMatcher.NO_MATCH) {
+            return loadResourceFromContactUri(uri, contentResolver);
+        }
         return contentResolver.openInputStream(uri);
     }
 
     @Override
     protected void close(InputStream data) throws IOException {
         data.close();
     }
+
+    private InputStream loadResourceFromContactUri(Uri uri, ContentResolver contentResolver)
+            throws FileNotFoundException {
+        Uri contactUri = uri;
+        int matchedUri = URI_MATCHER.match(contactUri);
+
+        switch (matchedUri) {
+            case ID_LOOKUP:
+            case ID_CONTACT:
+                // If it was a Lookup uri then resolve it first, then continue loading the contact uri.
+                if (matchedUri == ID_LOOKUP) {
+                    contactUri = ContactsContract.Contacts.lookupContact(contentResolver, contactUri);
+                    if (contactUri == null) {
+                        throw new FileNotFoundException("Contact cannot be found");
+                    }
+                }
+                if (SDK_INT < ICE_CREAM_SANDWICH) {
```

Comment:
I did it this way because Lint complained about this, if I moved this block into another method Lint will complain and the build will fail with: 
**NewApi: Calling new methods on older versions**

So the only way avoiding this error was doing it that way, if you have another way for it then I'm all ears of course.